### DISCUSSION TITLE
how to Open DrawerHost  outside it ?

### DISCUSSION THREAD

**Discussion by after1990s** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2909] [Q&A] [answered]
how to fill _________ get same behavior as btn0.

    <Button
        Command="{x:Static materialDesign:DrawerHost.OpenDrawerCommand}"
        CommandParameter="{x:Static ________}">
    <materialDesign:DrawerHost x:Name="DrawerHost">
        <materialDesign:DrawerHost.LeftDrawerContent>
                <Button/>
        </materialDesign:DrawerHost.LeftDrawerContent>
        
        <Button
            x:Name="btn0"
            Command="{x:Static materialDesign:DrawerHost.OpenDrawerCommand}"
            CommandParameter="{x:Static Dock.Left}"
            Content="{materialDesign:PackIcon Kind=ArrowLeft}">
    </materialDesign:DrawerHost>

**Comment by nicolaihenriksen** [comment, marked as answer]
@after1990s I think what you're trying to accomplish is to activate the drawer from an element located outside of the `DrawerHost`? For that, you could use the `CommandTarget` attribute to target the `DrawerHost` (or any other element inside of it) by its name:

```xaml
<Button
    Command="{x:Static materialDesign:DrawerHost.OpenDrawerCommand}"
    CommandParameter="{x:Static Dock.Left}"
    CommandTarget="{Binding ElementName=DrawerHost}"
    Content="{materialDesign:PackIcon Kind=ArrowLeft}"/>

<materialDesign:DrawerHost x:Name="DrawerHost">
    <materialDesign:DrawerHost.LeftDrawerContent>
            <Button/>
    </materialDesign:DrawerHost.LeftDrawerContent>
    
    <Button
        x:Name="btn0"
        Command="{x:Static materialDesign:DrawerHost.OpenDrawerCommand}"
        CommandParameter="{x:Static Dock.Left}"
        Content="{materialDesign:PackIcon Kind=ArrowLeft}">
</materialDesign:DrawerHost>
```

With this, pressing the top `Button` should have the same effect as pressing the "btn0" `Button`.
